test/integration/stats_integration_test.cc

```diff
@@ -210,6 +210,7 @@ TEST_P(ClusterMemoryTestRunner, MemoryLargeClusterSizeWithStats) {
   // 2019/06/29  7364     45685       46000   combine 2 levels of stat ref-counting into 1
   // 2019/06/30  7428     42742       43000   remove stats multiple inheritance, inline HeapStatData
   // 2019/07/06  7477     42742       43000   fork gauge representation to drop pending_increment_
+  // 2019/07/15  7555     42806       43000   static link libstdc++ in tests
```

Comment:
How did test behavior change in the PR? I thought we were just removing wrappers.

Reply:
This changes how we link libstdc++ in tests, previously, libstdc++ is dynamically linked for tests as `envoy_static_link_libstdcpp_linkopts` isn't added to `envoy_cc_test` linkopts. With this PR tests are linked with libstdc++ too so the number changed. Current master with following patch produces same number:

```
diff --git a/bazel/envoy_test.bzl b/bazel/envoy_test.bzl
index ea564d619..c502fa007 100644
--- a/bazel/envoy_test.bzl
+++ b/bazel/envoy_test.bzl
@@ -150,7 +150,7 @@ def envoy_cc_test(
     native.cc_test(
         name = name,
         copts = envoy_copts(repository, test = True) + copts,
-        linkopts = _envoy_test_linkopts(),
+        linkopts = _envoy_test_linkopts() + envoy_static_link_libstdcpp_linkopts(),
         linkstatic = envoy_linkstatic(),
         malloc = tcmalloc_external_dep(repository),
         deps = [
```

Reply:
Do we need to do this at the same time as this PR? If so, please update the top-level GH comment so we can also explain that this is happening.

Reply:
Yes because BAZEL_LINKLIBS is toolchain configuration so applies to all binary. Updated description.